cse-lib/network-processing/src/test/java/com/farao_community/farao/cse/network_processing/busbar_change/BusBarChangeProcessorTest.java

```diff
@@ -190,4 +187,64 @@ void test3NodeCase() {
         );
         assertEquals(switchPairs, bbcs.getSwitchPairs());
     }
+
+    @Test
+    void testTwoRasOnTwoEndsOfSameLine() {
+        setUp("BaseNetwork_twoBusBarsOnSameLine.uct", "BusBarChange_twoBusBarsOnSameLine.xml");
+        TestUtils.assertNetworksAreEqual(network, "ModifiedNetwork_twoBusBarsOnSameLine.uct", getClass());
+        assertEquals(3, busBarChangeSwitchesSet.size());
+        BusBarChangeSwitches bbcs = busBarChangeSwitchesSet.stream().filter(busBarChangeSwitches -> busBarChangeSwitches.getRemedialActionId().equals("RA_1")).findAny().orElseThrow();
+        Set<SwitchPairId> switchPairs = Set.of(
+            new SwitchPairId("BBE1AA1Z BBE1AA11 1", "BBE1AA1Z BBE1AA12 1")
+        );
+        assertEquals(switchPairs, bbcs.getSwitchPairs());
+        bbcs = busBarChangeSwitchesSet.stream().filter(busBarChangeSwitches -> busBarChangeSwitches.getRemedialActionId().equals("RA_2")).findAny().orElseThrow();
+        switchPairs = Set.of(
+            new SwitchPairId("BBE2AA1Z BBE2AA11 1", "BBE2AA1Z BBE2AA12 1")
+        );
+        assertEquals(switchPairs, bbcs.getSwitchPairs());
+        bbcs = busBarChangeSwitchesSet.stream().filter(busBarChangeSwitches -> busBarChangeSwitches.getRemedialActionId().equals("RA_1_inverted")).findAny().orElseThrow();
+        switchPairs = Set.of(
+            new SwitchPairId("BBE1AA1Z BBE1AA12 1", "BBE1AA1Z BBE1AA11 1")
+        );
+        assertEquals(switchPairs, bbcs.getSwitchPairs());
+
+        assertTrue(network.getIdentifiable("BBE1AA1Z BBE1AA11 1") instanceof Switch);
+        assertFalse(((Switch) network.getIdentifiable("BBE1AA1Z BBE1AA11 1")).isOpen());
+
+        assertTrue(network.getIdentifiable("BBE1AA1Z BBE1AA12 1") instanceof Switch);
+        assertTrue(((Switch) network.getIdentifiable("BBE1AA1Z BBE1AA12 1")).isOpen());
+
+        assertTrue(network.getIdentifiable("BBE2AA1Z BBE2AA11 1") instanceof Switch);
+        assertFalse(((Switch) network.getIdentifiable("BBE2AA1Z BBE2AA11 1")).isOpen());
+
+        assertTrue(network.getIdentifiable("BBE2AA1Z BBE2AA12 1") instanceof Switch);
+        assertTrue(((Switch) network.getIdentifiable("BBE2AA1Z BBE2AA12 1")).isOpen());
+
+        assertNull(network.getIdentifiable("BBE1AA11 BBE2AA11 1"));
+        assertNull(network.getIdentifiable("BBE1AA11 BBE2AA1Z 1"));
+        assertNull(network.getIdentifiable("BBE1AA1Z BBE2AA11 1"));
+        assertTrue(network.getIdentifiable("BBE1AA1Z BBE2AA1Z 1") instanceof Line);
+    }
+
+    @Test
+    void testTwoRasOnTwoEndsOfSameLine2() {
```

Comment:
Does this second test, tests anything else than the first one ?

Reply:
it's a bit more complex with different lines affected in each RA. in the first one only one line is affected

Reply:
alright :)